src/submission/submission-to-clinical/merge-submission.ts

```diff
@@ -18,150 +18,170 @@
+ * the existing donor data, and returns a new list of donors with the submission data added in.
+ * @param activeSubmission
+ * @param donors
+ * @returns
+ */
 export const mergeActiveSubmissionWithDonors = async (
-  activeSubmission: DeepReadonly<ActiveClinicalSubmission>,
-  donors: readonly DeepReadonly<Donor>[],
+	activeSubmission: DeepReadonly<ActiveClinicalSubmission>,
+	donors: readonly DeepReadonly<Donor>[],
 ) => {
-  const updatedDonors = _.cloneDeep(donors as Donor[]);
-  for (const entityType in activeSubmission.clinicalEntities) {
-    const entityData = activeSubmission.clinicalEntities[entityType];
-
-    // Find the donor that matches each record, and update the entity within that donor
-    entityData.records.forEach(record => {
-      const donorId = record[DonorFieldsEnum.submitter_donor_id];
-      const donor = _.find(updatedDonors, ['submitterId', donorId]);
-      if (!donor) {
-        throw new Errors.StateConflict(
-          `Donor ${donorId} has not been registered but is part of the activeSubmission, merge cannot be completed.`,
-        );
-      }
-      // update clinical info in clinical object
-      let entityWithUpdatedInfo: ClinicalEntity | Donor | undefined = { clinicalInfo: {} };
-      switch (entityType) {
-        case ClinicalEntitySchemaNames.DONOR:
-          entityWithUpdatedInfo = updateDonorInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.SPECIMEN:
-          entityWithUpdatedInfo = updateSpecimenInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.PRIMARY_DIAGNOSIS:
-          entityWithUpdatedInfo = updatePrimaryDiagnosisInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.FAMILY_HISTORY:
-          entityWithUpdatedInfo = updateFamilyHistoryInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.EXPOSURE:
-          entityWithUpdatedInfo = updateExposureInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.COMORBIDITY:
-          entityWithUpdatedInfo = updateComorbidityInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.BIOMARKER:
-          entityWithUpdatedInfo = updateBiomarkerInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.TREATMENT:
-          entityWithUpdatedInfo = updateOrAddTreatementInfo(donor, record);
-          break;
-        case ClinicalEntitySchemaNames.FOLLOW_UP:
-          entityWithUpdatedInfo = updateOrAddFollowUpInfo(donor, record);
-          break;
-        case ClinicalTherapySchemaNames.find(tsn => tsn === entityType):
-          entityWithUpdatedInfo = updateOrAddTherapyInfoInDonor(donor, record, entityType, true);
-          break;
-        default:
-          throw new Error(`Entity ${entityType} not implemented yet`);
-      }
-
-      // update clinical entity stats
-      updateDonorStatsFromSubmissionCommit(donor, entityType);
```

Comment:
This has been removed. The merge-submission code no longer recalculates core completion. This is done separately by the code that is submitting the data. merge-submission now is only updating the clinical entity data in the donor document.